rerun_cpp/src/rerun/recording_stream.cpp

```diff
@@ -91,6 +94,41 @@ namespace rerun {
         rr_recording_stream_flush_blocking(_id);
     }
 
+    Error RecordingStream::try_log_components(
+        const char* entity_path, const std::vector<AnonymousComponentList> component_lists
+    ) {
+        if (component_lists.size() == 0) {
+            return Error::ok();
+        }
+
+        auto num_instances = component_lists[0].size;
```

Comment:
I don't feel too good about this... In rust we just take `num_instances` as a parameter which is far from great but still better than this imo.